# Exploring Ebay Car Sales Data

Data set contains information about used cars scrapped from German eBay website. It is available under https://www.kaggle.com/orgesleka/used-cars-database/data.

Goal of the project is to clean the data and make analysis from used car listings.


In [26]:
import pandas as pd
autos = pd.read_csv('autos.csv',encoding='Latin-1')

In [27]:
autos.info()
autos.head(10)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 50000 entries, 0 to 49999
Data columns (total 20 columns):
dateCrawled            50000 non-null object
name                   50000 non-null object
seller                 50000 non-null object
offerType              50000 non-null object
price                  50000 non-null object
abtest                 50000 non-null object
vehicleType            44905 non-null object
yearOfRegistration     50000 non-null int64
gearbox                47320 non-null object
powerPS                50000 non-null int64
model                  47242 non-null object
odometer               50000 non-null object
monthOfRegistration    50000 non-null int64
fuelType               45518 non-null object
brand                  50000 non-null object
notRepairedDamage      40171 non-null object
dateCreated            50000 non-null object
nrOfPictures           50000 non-null int64
postalCode             50000 non-null int64
lastSeen               50000 non-null obj

dateCrawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   
5  2016-03-21 13:47:45  Chrysler_Grand_Voyager_2.8_CRD_Aut.Limited_Sto...   
6  2016-03-20 17:55:21  VW_Golf_III_GT_Special_Electronic_Green_Metall...   
7  2016-03-16 18:55:19                               Golf_IV_1.9_TDI_90PS   
8  2016-03-22 16:51:34                                         Seat_Arosa   
9  2016-03-16 13:47:02          Renault_Megane_Scenic_1.6e_RT_Klimaanlage   

   seller offerType   price   abtest vehicleType  yearOfRegistration  \
0  privat   Angebot  $5,000  control         bus                2004   
1  privat   Angebot  $8,500  control   limousine                1997   
2  privat   Angebot  $8,990     test   limousine                2009   
3  privat   Angebot  $4,350  control  kleinwagen                2007   
4  privat   Angebot  $1,350     test       kombi                2003   
5  privat   Angebot  $7,900     test         bus                2006   
6  privat   Angebot    $300     test   limousine                1995   
7  privat   Angebot  $1,990  control   limousine                1998   
8  privat   Angebot    $250     test         NaN                2000   
9  privat   Angebot    $590  control         bus                1997   

     gearbox  powerPS    model   odometer  monthOfRegistration fuelType  \
0    manuell      158   andere  150,000km                    3      lpg   
1  automatik      286      7er  150,000km                    6   benzin   
2    manuell      102     golf   70,000km                    7   benzin   
3  automatik       71   fortwo   70,000km                    6   benzin   
4    manuell        0    focus  150,000km                    7   benzin   
5  automatik      150  voyager  150,000km                    4   diesel   
6    manuell       90     golf  150,000km                    8   benzin   
7    manuell       90     golf  150,000km                   12   diesel   
8    manuell        0    arosa  150,000km                   10      NaN   
9    manuell       90   megane  150,000km                    7   benzin   

        brand notRepairedDamage          dateCreated  nrOfPictures  \
0     peugeot              nein  2016-03-26 00:00:00             0   
1         bmw              nein  2016-04-04 00:00:00             0   
2  volkswagen              nein  2016-03-26 00:00:00             0   
3       smart              nein  2016-03-12 00:00:00             0   
4        ford              nein  2016-04-01 00:00:00             0   
5    chrysler               NaN  2016-03-21 00:00:00             0   
6  volkswagen               NaN  2016-03-20 00:00:00             0   
7  volkswagen              nein  2016-03-16 00:00:00             0   
8        seat              nein  2016-03-22 00:00:00             0   
9     renault              nein  2016-03-16 00:00:00             0   

   postalCode             lastSeen  
0       79588  2016-04-06 06:45:54  
1       71034  2016-04-06 14:45:08  
2       35394  2016-04-06 20:15:37  
3       33729  2016-03-15 03:16:28  
4       39218  2016-04-01 14:38:50  
5       22962  2016-04-06 09:45:21  
6       31535  2016-03-23 02:48:59  
7       53474  2016-04-07 03:17:32  
8        7426  2016-03-26 18:18:10  
9       15749  2016-04-06 10:46:35

From above information we can notice that:
- columns names use camelcase
- car name column does not contain white spaces (it has been separated by underscore sign)
- dataset contain German names
- 5 columns are numerics, other are strings
- 5 columns contains null values (vehicleType, gearbox, model, fuelType, notRepairedDamage)
- price value is in string format (it contains $ sign)
- there are three column with date
- date format is in YYYY-MM-DD HH:mm:ss

In [28]:
import re
def convert_to_snake(name):
    s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
    return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1).lower()
def clean_col(column_name):
    column_name = column_name.replace("yearOfRegistration","registration_year")
    column_name = column_name.replace("monthOfRegistration","registration_month")
    column_name = column_name.replace("notRepairedDamage","unrepaired_damage")
    column_name = column_name.replace("dateCreated","ad_created")
    column_name = convert_to_snake(column_name)
    return column_name

cleaned_cols = [] 
for name in autos.columns:
    cleaned_column = clean_col(name)
    cleaned_cols.append(cleaned_column)

autos.columns = cleaned_cols
autos.head()

date_crawled                                               name  \
0  2016-03-26 17:47:46                   Peugeot_807_160_NAVTECH_ON_BOARD   
1  2016-04-04 13:38:56         BMW_740i_4_4_Liter_HAMANN_UMBAU_Mega_Optik   
2  2016-03-26 18:57:24                         Volkswagen_Golf_1.6_United   
3  2016-03-12 16:58:10  Smart_smart_fortwo_coupe_softouch/F1/Klima/Pan...   
4  2016-04-01 14:38:50  Ford_Focus_1_6_Benzin_TÜV_neu_ist_sehr_gepfleg...   

   seller offer_type   price   abtest vehicle_type  registration_year  \
0  privat    Angebot  $5,000  control          bus               2004   
1  privat    Angebot  $8,500  control    limousine               1997   
2  privat    Angebot  $8,990     test    limousine               2009   
3  privat    Angebot  $4,350  control   kleinwagen               2007   
4  privat    Angebot  $1,350     test        kombi               2003   

     gearbox  power_ps   model   odometer  registration_month fuel_type  \
0    manuell       158  andere  150,000km                   3       lpg   
1  automatik       286     7er  150,000km                   6    benzin   
2    manuell       102    golf   70,000km                   7    benzin   
3  automatik        71  fortwo   70,000km                   6    benzin   
4    manuell         0   focus  150,000km                   7    benzin   

        brand unrepaired_damage           ad_created  nr_of_pictures  \
0     peugeot              nein  2016-03-26 00:00:00               0   
1         bmw              nein  2016-04-04 00:00:00               0   
2  volkswagen              nein  2016-03-26 00:00:00               0   
3       smart              nein  2016-03-12 00:00:00               0   
4        ford              nein  2016-04-01 00:00:00               0   

   postal_code            last_seen  
0        79588  2016-04-06 06:45:54  
1        71034  2016-04-06 14:45:08  
2        35394  2016-04-06 20:15:37  
3        33729  2016-03-15 03:16:28  
4        39218  2016-04-01 14:38:50

Due to Python prefer snakecase above code convert all the columns name from camelcase to snakecase. Also, some of the columns have to be more decriptive (ex. name dateCreate is unclear)  

In [29]:
autos.describe(include='all')

date_crawled         name  seller offer_type  price abtest  \
count                 50000        50000   50000      50000  50000  50000   
unique                48213        38754       2          2   2357      2   
top     2016-03-30 17:37:35  Ford_Fiesta  privat    Angebot     $0   test   
freq                      3           78   49999      49999   1421  25756   
mean                    NaN          NaN     NaN        NaN    NaN    NaN   
std                     NaN          NaN     NaN        NaN    NaN    NaN   
min                     NaN          NaN     NaN        NaN    NaN    NaN   
25%                     NaN          NaN     NaN        NaN    NaN    NaN   
50%                     NaN          NaN     NaN        NaN    NaN    NaN   
75%                     NaN          NaN     NaN        NaN    NaN    NaN   
max                     NaN          NaN     NaN        NaN    NaN    NaN   

       vehicle_type  registration_year  gearbox      power_ps  model  \
count         44905       50000.000000    47320  50000.000000  47242   
unique            8                NaN        2           NaN    245   
top       limousine                NaN  manuell           NaN   golf   
freq          12859                NaN    36993           NaN   4024   
mean            NaN        2005.073280      NaN    116.355920    NaN   
std             NaN         105.712813      NaN    209.216627    NaN   
min             NaN        1000.000000      NaN      0.000000    NaN   
25%             NaN        1999.000000      NaN     70.000000    NaN   
50%             NaN        2003.000000      NaN    105.000000    NaN   
75%             NaN        2008.000000      NaN    150.000000    NaN   
max             NaN        9999.000000      NaN  17700.000000    NaN   

         odometer  registration_month fuel_type       brand unrepaired_damage  \
count       50000        50000.000000     45518       50000             40171   
unique         13                 NaN         7          40                 2   
top     150,000km                 NaN    benzin  volkswagen              nein   
freq        32424                 NaN     30107       10687             35232   
mean          NaN            5.723360       NaN         NaN               NaN   
std           NaN            3.711984       NaN         NaN               NaN   
min           NaN            0.000000       NaN         NaN               NaN   
25%           NaN            3.000000       NaN         NaN               NaN   
50%           NaN            6.000000       NaN         NaN               NaN   
75%           NaN            9.000000       NaN         NaN               NaN   
max           NaN           12.000000       NaN         NaN               NaN   

                 ad_created  nr_of_pictures   postal_code            last_seen  
count                 50000         50000.0  50000.000000                50000  
unique                   76             NaN           NaN                39481  
top     2016-04-03 00:00:00             NaN           NaN  2016-04-07 06:17:27  
freq                   1946             NaN           NaN                    8  
mean                    NaN             0.0  50813.627300                  NaN  
std                     NaN             0.0  25779.747957                  NaN  
min                     NaN             0.0   1067.000000                  NaN  
25%                     NaN             0.0  30451.000000                  NaN  
50%                     NaN             0.0  49577.000000                  NaN  
75%                     NaN             0.0  71540.000000                  NaN  
max                     NaN             0.0  99998.000000                  NaN

Above statistics shows that:
- Nr of pictures does not contain any useful information (all the values are zeros) so it can be dropped.
- column registration_year contains some values which are not correct (like 1000.0 and 9999.0) so it is need to investigate some deeper.
- column power_ps contains some values which are not correct (like 0.0) which need deeper investigation
- price needs to be converted to float type. Some of the price values contains $ sign what needs to be cleaned
- odometer rows needs to be converted to float type. Some of the price values contains km what needs to be cleaned.

In [30]:
print(autos["price"].unique())
print(autos["odometer"].unique())

['$5,000' '$8,500' '$8,990' ... '$385' '$22,200' '$16,995']
['150,000km' '70,000km' '50,000km' '80,000km' '10,000km' '30,000km'
 '125,000km' '90,000km' '20,000km' '60,000km' '5,000km' '100,000km'
 '40,000km']


In [31]:
autos["price"] = (autos["price"]
                  .str.replace("$","")
                  .str.replace(",","")
                  .astype(int)
                 )
autos["odometer"] = (autos["odometer"]
                  .str.replace("km","")
                  .str.replace(",","")
                  .astype(int)
                 )
autos.rename({"odometer":"odometer_km"}, axis='columns',inplace=True)

In [32]:
top_prices_counts = autos["price"].value_counts().sort_index().head(10)
tail_prices_counts = autos["price"].value_counts().sort_index(ascending=False).head(10)
unique_prices = autos["price"].unique().shape
print('''
{}
{} 
'''.format(unique_prices,top_prices_counts,tail_prices_counts))
print('''*ODOMETER all value with counts*
{}'''
      .format(autos["odometer_km"].value_counts()))


(2357,)
0     1421
1      156
2        3
3        1
5        2
8        1
9        1
10       7
11       2
12       3
Name: price, dtype: int64 

*ODOMETER all value with counts*
150000    32424
125000     5170
100000     2169
90000      1757
80000      1436
70000      1230
60000      1164
50000      1027
5000        967
40000       819
30000       789
20000       784
10000       264
Name: odometer_km, dtype: int64


According to statistics about price and odometer there is need to clean some prices which don't look not realistics. There are a lot of with price 0\$. I left only cars which costs less than 7M\$ and more than 100\$ 

In [33]:
autos_price_clean = autos[autos["price"].between(100,7000000)]
print(autos_price_clean["price"].describe())
print(autos_price_clean["price"].unique().shape)

count    4.823000e+04
mean     6.125042e+03
std      2.286169e+04
min      1.000000e+02
25%      1.250000e+03
50%      3.000000e+03
75%      7.499000e+03
max      3.890000e+06
Name: price, dtype: float64
(2315,)


Nearly 2000 rows have been deleted, there was 52 unique values which was between 100\$ and 7M\$ 

In [34]:
date_crawled_part = autos_price_clean["date_crawled"].str[:10]
ad_created_part = autos_price_clean["ad_created"].str[:7]
last_seen_part = autos_price_clean["last_seen"].str[:10]
print(date_crawled_part.value_counts(normalize=True,dropna=False).sort_index())
print(ad_created_part.value_counts(normalize=True,dropna=False).sort_index())
print(last_seen_part.value_counts(normalize=True,dropna=False).sort_index())

2016-03-05    0.025358
2016-03-06    0.014037
2016-03-07    0.036056
2016-03-08    0.033174
2016-03-09    0.033009
2016-03-10    0.032283
2016-03-11    0.032594
2016-03-12    0.036906
2016-03-13    0.015675
2016-03-14    0.036658
2016-03-15    0.034315
2016-03-16    0.029463
2016-03-17    0.031516
2016-03-18    0.012897
2016-03-19    0.034729
2016-03-20    0.037798
2016-03-21    0.037217
2016-03-22    0.032905
2016-03-23    0.032283
2016-03-24    0.029442
2016-03-25    0.031495
2016-03-26    0.032304
2016-03-27    0.031122
2016-03-28    0.034957
2016-03-29    0.034128
2016-03-30    0.033734
2016-03-31    0.031847
2016-04-01    0.033693
2016-04-02    0.035600
2016-04-03    0.038607
2016-04-04    0.036575
2016-04-05    0.013062
2016-04-06    0.003172
2016-04-07    0.001389
Name: date_crawled, dtype: float64
2015-06    0.000021
2015-08    0.000021
2015-09    0.000021
2015-11    0.000021
2015-12    0.000041
2016-01    0.000249
2016-02    0.001265
2016-03    0.837321
2016-04    0.161041
Nam

The code above extract date part from column with date. Then it is printed all values with frequency of appearance from earliest date to latest.

It is visible that last_seen date and date_crawled is in the same range (1 month period). Most ad_created is from march 2016 values. There is no null values.  

In [35]:
autos_price_clean["registration_year"].describe()

count    48230.000000
mean      2004.729795
std         87.892058
min       1000.000000
25%       1999.000000
50%       2004.000000
75%       2008.000000
max       9999.000000
Name: registration_year, dtype: float64

Description of registration_year shows that it contains values which are incorrect (1000 and 9999). It is needed to be replaced.

The first car have been created in 1886. Data set have been created in 2016, so it is impossible that there is cars which have been registrated after 2016.So I assumed that registration year before 1900 and after 2016 is incorrect.


In [36]:
autos_cleaned_registration = (
    autos_price_clean[autos_price_clean["registration_year"]
                     .between(1900,2016)])
print(autos_cleaned_registration["registration_year"]
      .value_counts(normalize=True))

2000    0.066957
2005    0.062794
1999    0.062125
2004    0.058221
2003    0.058091
2006    0.057574
2001    0.056711
2002    0.053432
1998    0.050477
2007    0.049031
2008    0.047672
2009    0.044890
1997    0.041525
2011    0.034902
2010    0.034234
1996    0.029229
2012    0.028215
2016    0.025929
1995    0.025735
2013    0.017279
2014    0.014280
1994    0.013504
1993    0.009060
2015    0.008197
1992    0.007938
1991    0.007291
1990    0.007162
1989    0.003689
1988    0.002869
1985    0.002028
          ...   
1966    0.000475
1976    0.000453
1969    0.000410
1975    0.000388
1965    0.000367
1964    0.000259
1963    0.000173
1961    0.000129
1959    0.000129
1962    0.000086
1956    0.000086
1937    0.000086
1958    0.000086
1955    0.000043
1954    0.000043
1957    0.000043
1951    0.000043
1910    0.000043
1934    0.000043
1941    0.000043
1927    0.000022
1929    0.000022
1950    0.000022
1931    0.000022
1948    0.000022
1938    0.000022
1939    0.000022
1953    0.0000

10 most often registered years are from 1998 to 2007. It is about 50% of all the cars. Registration date before 1980  is very rare.

In [37]:

brand_distribution = autos_cleaned_registration["brand"].value_counts(normalize=True)
print(brand_distribution,brand_distribution.shape)
brands_over_1_percent = brand_distribution[brand_distribution>0.02].index

volkswagen        0.211420
bmw               0.110186
opel              0.107231
mercedes_benz     0.096639
audi              0.086760
ford              0.069848
renault           0.047068
peugeot           0.029855
fiat              0.025605
seat              0.018249
skoda             0.016416
nissan            0.015337
mazda             0.015229
smart             0.014194
citroen           0.014043
toyota            0.012792
hyundai           0.010009
sonstige_autos    0.009578
volvo             0.009125
mini              0.008801
mitsubishi        0.008176
honda             0.007874
kia               0.007075
alfa_romeo        0.006666
porsche           0.006018
suzuki            0.005932
chevrolet         0.005673
chrysler          0.003516
dacia             0.002653
daihatsu          0.002502
jeep              0.002287
subaru            0.002114
land_rover        0.002114
saab              0.001661
jaguar            0.001532
daewoo            0.001488
trabant           0.001359
r

There is 40 different brands. Let's agregate some of the most popular ones, for example which have over 2% of the total values. It is good to see what are the brand average price.

In [38]:
brand_mean_prices = {}
for brand in brands_over_1_percent:
    brand_mean_price = autos_cleaned_registration.loc[
        autos_cleaned_registration["brand"]==brand,"price"].mean(axis=0)
    brand_mean_prices[brand] = brand_mean_price
print(brand_mean_prices)

{'fiat': 2836.8736310025274, 'ford': 4086.93421865349, 'audi': 9380.718547986076, 'peugeot': 3113.860549132948, 'opel': 3005.4960772480385, 'volkswagen': 5639.900316294255, 'bmw': 8621.729052466719, 'renault': 2496.070577451879, 'mercedes_benz': 8672.654241071428}


The most expensive car from listed above is audi, and the cheapest is fiat and renault. The top-3 most expensive cars are from Germany

In [39]:
brand_mean_mileages = {}
for brand in brands_over_1_percent:
    brand_mean_mileage = autos_cleaned_registration.loc[
        autos_cleaned_registration["brand"]==brand,"odometer_km"].mean(axis=0)
    brand_mean_mileages[brand] = brand_mean_mileage
print(brand_mean_mileages)
brand_mean_registrations = {}
for brand in brands_over_1_percent:
    brand_mean_registration = autos_cleaned_registration.loc[
        autos_cleaned_registration["brand"]==brand,"registration_year"].mean(axis=0)
    brand_mean_registrations[brand] = brand_mean_registration
print(brand_mean_registrations)

{'fiat': 116950.29486099411, 'ford': 124277.33168622607, 'audi': 129245.40029835903, 'peugeot': 127127.8901734104, 'opel': 129384.42969221485, 'volkswagen': 128804.2036526885, 'bmw': 132698.7079091621, 'renault': 128281.3932172319, 'mercedes_benz': 131025.66964285714}
{'fiat': 2003.267902274642, 'ford': 2002.7906114885732, 'audi': 2004.1718050721033, 'peugeot': 2003.7796242774566, 'opel': 2002.26554013277, 'volkswagen': 2002.4793388429753, 'bmw': 2003.0432654659357, 'renault': 2002.9106324472962, 'mercedes_benz': 2002.052232142857}


In [40]:
bmp_series = pd.Series(brand_mean_prices)
bmm_series = pd.Series(brand_mean_mileages)
bmr_series = pd.Series(brand_mean_registrations)
df_agg_brand = pd.DataFrame(bmp_series,columns=["brand_mean_price"])
df_agg_brand["brand_mean_mileages"] = bmm_series
df_agg_brand["brand_registration_year"] = bmr_series
df_agg_brand

brand_mean_price  brand_mean_mileages  brand_registration_year
audi                9380.718548        129245.400298              2004.171805
bmw                 8621.729052        132698.707909              2003.043265
fiat                2836.873631        116950.294861              2003.267902
ford                4086.934219        124277.331686              2002.790611
mercedes_benz       8672.654241        131025.669643              2002.052232
opel                3005.496077        129384.429692              2002.265540
peugeot             3113.860549        127127.890173              2003.779624
renault             2496.070577        128281.393217              2002.910632
volkswagen          5639.900316        128804.203653              2002.479339

There is not visible correlation between brand price and mlieages. Fiat has mean mileage the smallest and the price is the one of the cheapest. Renault is the cheapest one and the mileage is similar as volkswagen which is two times more expensive.

It is visible that avearage registration year is latest for audi. Difference is about one year in comparision to others cars. However, mercedes_benz mean is the oldest and the price is one of the highest.  

In [41]:
ger_eng_fuel_type = {
    "benzin":"gasoline","diesel":"diesel","lpg":"lpg",
    "cng":"cng","hybrid":"hybrid","elektro":"electric",
    "andere":"others"
}
autos_cleaned_registration["fuel_type"].replace(ger_eng_fuel_type,inplace=True)

fuel_types = autos_cleaned_registration["fuel_type"].value_counts(normalize=True).index
fuel_types_mean_prices = {}

for fuel_type in fuel_types:
    print(fuel_type)
    fuel_types_mean_price = autos_cleaned_registration.loc[
        autos_cleaned_registration["fuel_type"]==fuel_type,"price"].mean(axis=0)
    fuel_types_mean_prices[fuel_type] = fuel_types_mean_price
print(fuel_types_mean_prices)

gasoline
diesel
lpg
cng
hybrid
electric
others
{'hybrid': 14346.027027027027, 'diesel': 8682.989082346225, 'others': 3185.6428571428573, 'gasoline': 5188.389479988726, 'electric': 26089.444444444445, 'lpg': 4334.817901234568, 'cng': 4930.785714285715}


/dataquest/system/env/python3/lib/python3.4/site-packages/pandas/core/generic.py:4619: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  self._update_inplace(new_data)


Cars with the lpg are the cheapest one. The most expensive are hybrid and electric cars. Gasoline cars are twice cheaper than diesel cars

In [42]:
fuel_types_in_brand = {}
for brand in brands_over_1_percent:
    idx_brand_fuel_type = ((autos_cleaned_registration["brand"]==brand))
    nr_ft_in_brand = autos_cleaned_registration.loc[idx_brand_fuel_type,"fuel_type"].value_counts()
    fuel_types_in_brand[brand] = nr_ft_in_brand
print(fuel_types_in_brand)

{'fiat': gasoline    864
diesel      185
cng          13
lpg          12
Name: fuel_type, dtype: int64, 'ford': gasoline    2037
diesel       893
lpg           28
others         1
Name: fuel_type, dtype: int64, 'audi': gasoline    1928
diesel      1847
lpg           49
hybrid         2
cng            1
Name: fuel_type, dtype: int64, 'peugeot': gasoline    964
diesel      286
lpg          17
electric      1
Name: fuel_type, dtype: int64, 'opel': gasoline    3548
diesel       897
lpg           70
cng           11
others         1
hybrid         1
Name: fuel_type, dtype: int64, 'volkswagen': gasoline    5456
diesel      3529
lpg           80
cng           33
hybrid         1
Name: fuel_type, dtype: int64, 'bmw': gasoline    2837
diesel      1890
lpg           92
cng            3
electric       2
hybrid         1
Name: fuel_type, dtype: int64, 'renault': gasoline    1562
diesel       419
lpg           16
electric       5
hybrid         1
cng            1
others         1
Name: fuel_type, d

Regardless of brand most of the cars use gasoline. There is only few examples of cars which are hybrid or electric. It is noticable that there is higher ratio diesel to gasoline in audi and mercedes_benz brands. If we look at relation fuel type to  price the diesel cars are twice more expensive than gasoline.

In [51]:
most_common_model = ()
brand_most_common_model = {}
for brand in brands_over_1_percent:
    model_dist = (autos_cleaned_registration.loc[
        autos_cleaned_registration["brand"] == brand,"model"]
                           .value_counts())
    if model_dist.shape[0] > 0:    
        brand_most_common_model[brand] = model_dist.index[0]
    #if most_common_model_brand  > most_common_model_brand.values:
print(brand_most_common_model)

{'fiat': 'punto', 'ford': 'focus', 'audi': 'a4', 'peugeot': '2_reihe', 'opel': 'corsa', 'volkswagen': 'golf', 'bmw': '3er', 'renault': 'twingo', 'mercedes_benz': 'c_klasse'}
